viewsb/descriptor.py

```diff
@@ -103,7 +103,10 @@ def decode_data_as_descriptor(cls, data, use_pretty_names=True, parent=None, sub
         # Say that five times fast.
 
         # FIXME: memoize this?
-        parsed_data = cls.BINARY_FORMAT.parse(data)
+        try:
+            parsed_data = cls.BINARY_FORMAT.parse(data)
+        except:
```

Comment:
This is a footgun, any weird issues that may happen during this call won't be reported, the bare except will also catch keyboard interrupts and other similar exceptions. Please consider specifying the specific error classes we are expecting, if that is not possible use `except Exception` to avoid catching `KeyboardInterrupt`.

Reply:
I would have been more specific with the except, but I couldn't (at least on stream) locate any documentation for more specific exceptions to catch from construct - this prevents actual crashes, with a reasonable fall-back behaviour.

If nobody (including me) finds more suitable exceptions, I'll change it to `except Exception` as suggested.

Reply:
The base class for construct errors is `construct.core.ConstructError`; might as well change it to that?

Reply:
Thank you for the clarification. I will update the current version of this fix (reviewed below) to use ConstructError then.